feature/environment

vancouver29 opened the pull request:
the structure of Object Invironment is like this ?

vancouver29 committed:
Update types.rs

the structure of Object Invironment is like this ?

johnxnguyen commented on src/physics/types.rs:
```diff
@@ -142,6 +142,26 @@ impl Field {
+
+// Invironment ///////////////////////////////////////////////////////////////
+
+pub struct Invironment {
```
There is a typo: Environment instead of Invironment.

johnxnguyen commented on src/physics/types.rs:
```diff
@@ -142,6 +142,26 @@ impl Field {
+            
+        }
+    }
+
```
Try to keep the style consistent with the rest of the file. You should delete empty lines after method bodies. The same goes for before and after struct properties.

johnxnguyen commented on src/physics/types.rs:
```diff
@@ -142,6 +142,26 @@ impl Field {
+    }
+
+
+}
```
Also please add a new line between the implementation and the test header

johnxnguyen commented on src/physics/types.rs:
```diff
@@ -142,6 +142,26 @@ impl Field {
+pub struct Invironment {
+
+    pub bodies: HashMap<u32, Body>,
+    pub fields: HashMap<u32, Field>,
```
I would suggest to store the fields in a `Vec`, rather than a `HashMap`, there's no reason to keep in in a `HashMap`.

vancouver29 replied:
how about this : 
`pub fields: Vec<Field>, ` ?

vancouver29 committed:
Update types.rs

how about the list of body?

vancouver29 committed:
Update types.rs

Lines are fixed

johnxnguyen commented on src/physics/types.rs:
```diff
@@ -142,6 +142,29 @@ impl Field {
 
+
+
+// Invironment ///////////////////////////////////////////////////////////////
```
Still a typo here.

johnxnguyen commented on src/physics/types.rs:
```diff
@@ -142,6 +142,29 @@ impl Field {
+impl Environment {
+    pub fn update(&mut self) {
+
+        // update each field
```
There are still some empty lines here.

johnxnguyen reviewed (dismissed):
There are still some styling issues but otherwise looks good 👍

vancouver29 committed:
update/Environment

the bodies and their appropriate forces are created , then update the state of each field of fields by applying force on each of the bodies

vancouver29 committed:
update/Enviroment
